apps/fxc-front/src/app/components/2d/planner-element.ts

```diff
@@ -261,4 +294,38 @@ export class PlannerElement extends connect(store)(LitElement) {
   private wheelSpeed(e: WheelEvent): void {
     store.dispatch(e.deltaY > 0 ? incrementSpeed() : decrementSpeed());
   }
+
+  private handleScoreAction() {
+    if (!this.track || !this.scorer) {
+      return;
+    }
+    const track = this.track;
+    const points = track.lat.map((lat, index) => ({
+      lat,
+      lon: track.lon[index],
+      alt: track.alt[index],
+      timeSec: track.timeSec[index],
+    }));
+    this.scorer.score(points, this.league, ++this.scoringRequestId);
+  }
+
+  private handleScoringResult(result: ScoringResult) {
+    store.dispatch(setScore({ ...result, origin: ScoreOrigin.TRACK }));
+    if (this.track) {
+      const lastTimeSec = this.track.timeSec.at(-1);
+      const durationS = lastTimeSec === undefined ? 0 : lastTimeSec - this.track.timeSec[0];
```

Comment:
0 sounds wrong here.
you can take the if branch only if this.track.length > 1

Reply:
I changed to this (even if you change the test  with this.track.length > 1, this.track.timeSec.at(-1) will always return number|undefined).
```
    const lastTimeSec = this.currentTrack?.timeSec?.at(-1);
    const firstTimeSec = this.currentTrack?.timeSec?.at(0);
    if (lastTimeSec && firstTimeSec) {
      const durationS = lastTimeSec - firstTimeSec;
```
The downside is that `at(0)` is not really idiomatic. (this.currentTrack?.timeSec?[0] is not a valid syntax)

Reply:
You should still test `this.track.length > 1` and `const lastTimeSec = this.track.timeSec.at(-1)!;`

I think next TS release will improve the typings in this area but for now using `!` is fine